### PR TITLE
feat: add task to print all test source sets

### PR DESCRIPTION
**Pull request by jochil**
... for using it in cifuzz when collecting all tests during bundling

### CODE REVIEW

File: src/main/kotlin/com/code_intelligence/cifuzz/TestSetAccess.kt [hunk 1]
```diff
@@ -12,4 +12,5 @@ interface TestSetAccess {
     val mainClasses: FileCollection
     val mainSources: FileCollection
     val testTask: Provider<out Task>
+    val testSources:FileCollection
```
Comment (jjohannes) [inline src/main/kotlin/com/code_intelligence/cifuzz/TestSetAccess.kt:15]:
```suggestion
    val testSources: FileCollection
```
💄

File: src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * prints all the source test set paths to handle
+ * custom source sets inside cifuzz
+ */
+abstract class TestSourceSetsPrinter: DefaultTask() {
```
Comment (jjohannes) [inline src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt:12]:
```suggestion
abstract class TestSourceSetsPrinter : DefaultTask() {
```
💄

File: src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    @get:InputFiles
+    abstract val testSourceSets: ConfigurableFileCollection
+    @TaskAction
+    fun print(){
+        println("cifuzz.src-sets.test=${testSourceSets.asPath}")
```
Comment (jjohannes) [inline src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt:18]: `cifuzz.src-sets.test` might be confusing. If we refer to a _Source Set_ in the Gradle sense, we only have **one** source set that we select (which the user may configure via `ciFuzz.testSourceSet`). That source set can have multiple **source folders**. Also I would put the "test" first to be consistent with `cifuzz.test.classpath`.

How about:

`cifuzz.test.sources`

or

`cifuzz.test.source-folders`

File: src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * prints all the source test set paths to handle
+ * custom source sets inside cifuzz
+ */
+abstract class TestSourceSetsPrinter: DefaultTask() {
+
+    @get:InputFiles
+    abstract val testSourceSets: ConfigurableFileCollection
```
Comment (jjohannes) [inline src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt:15]:
```suggestion
    @get:InputFiles
    abstract val testSourceSets: ConfigurableFileCollection
    
```
💄 (add empty line)

File: src/main/kotlin/com/code_intelligence/cifuzz/android/AndroidTestSetAccess.kt [hunk 5]
```diff
@@ -48,4 +48,7 @@ class AndroidTestSetAccess(
 
     override val testTask: Provider<out Task>
         get() = project.provider { project.tasks.named("${testTaskPrefix()}${testComponent().name.capitalized()}") }.flatMap { it }
+
+    override val testSources: FileCollection
+        get() = TODO("Not yet implemented")
```
Comment (jjohannes) [inline src/main/kotlin/com/code_intelligence/cifuzz/android/AndroidTestSetAccess.kt:53]:
```suggestion
        get() = project.objects.fileCollection().from(testComponent().sources.java?.all, testComponent().sources.kotlin?.all)
```

...and possibly add a corresponding test to `MultiProjectAndroidTest`:

```
        val result = runner("cifuzzPrintTestSourceSets", "-q").build()

        assertEquals(TaskOutcome.SUCCESS, result.task(":app:cifuzzPrintTestSourceSets")?.outcome)
        assertThat(result.output, containsString(File("src/test${testedAndroidVariant().capitalized()}/").path))
        assertThat(result.output, containsString("cifuzz.src-sets.test="))
```

File: src/main/kotlin/com/code_intelligence/cifuzz/config/TaskConfigurations.kt [hunk 6]
```diff
@@ -75,6 +73,12 @@ private fun Project.registerPrintPackages(testSetAccess: TestSetAccess) {
     }
 }
 
+private fun Project.registerPrintTestSourceSets(testSetAccess: TestSetAccess) {
+    tasks.register("cifuzzPrintTestSourceSets", TestSourceSetsPrinter::class.java) { printTestSourceSets ->
```
Comment (jjohannes) [inline src/main/kotlin/com/code_intelligence/cifuzz/config/TaskConfigurations.kt:77]:
```suggestion
    tasks.register("cifuzzPrintTestSourceFolders", TestSourceSetsPrinter::class.java) { printTestSourceSets ->
```

...or `cifuzzPrintTestSources`

See my other comment about "source set" vs "source folders".

File: src/main/kotlin/com/code_intelligence/cifuzz/android/AndroidTestSetAccess.kt (hunk 5 above)
Comment (jochil) [reply src/main/kotlin/com/code_intelligence/cifuzz/android/AndroidTestSetAccess.kt]: great thanks

File: src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt (hunk 3 above)
Comment (jochil) [reply src/main/kotlin/com/code_intelligence/cifuzz/tasks/TestSourceSetsPrinter.kt]: you're right... `source-folders` is more specific

File: src/main/kotlin/com/code_intelligence/cifuzz/android/AndroidTestSetAccess.kt (hunk 5 above)
Comment (jochil) [reply src/main/kotlin/com/code_intelligence/cifuzz/android/AndroidTestSetAccess.kt]: great thanks

**Review by jjohannes — APPROVED** [review]

**Pull request merged**